dbms/src/Storages/DeltaMerge/Delta/ColumnFilePersistedSet.cpp

```diff
@@ -156,26 +130,15 @@ ColumnFilePersisteds ColumnFilePersistedSet::diffColumnFiles(const ColumnFiles &
     //       that this function is called under a for_update snapshot context.
     RUNTIME_CHECK(previous_column_files.size() <= getColumnFileCount());
 
-    // We check in the direction from the last level to the first level.
-    // In every level, we check from the begin to the last.
     auto it_1 = previous_column_files.begin();
-    auto level_it = persisted_files_levels.rbegin();
-    auto it_2 = level_it->begin();
+    auto it_2 = persisted_files.begin();
     bool check_success = true;
     if (likely(previous_column_files.size() <= persisted_files_count.load()))
     {
-        while (it_1 != previous_column_files.end() && level_it != persisted_files_levels.rend())
+        while (it_1 != previous_column_files.end() && it_2 != persisted_files.end())
         {
-            if (it_2 == level_it->end())
-            {
-                level_it++;
-                if (unlikely(level_it == persisted_files_levels.rend()))
-                    throw Exception("Delta Check head algorithm broken", ErrorCodes::LOGICAL_ERROR);
-                it_2 = level_it->begin();
-                continue;
-            }
             // We allow passing unflushed memtable files to `previous_column_files`, these heads will be skipped anyway.
-            if (!(*it_2)->mayBeFlushedFrom(&**it_1) && !(*it_1)->isSame(&**it_1))
+            if (!(*it_2)->mayBeFlushedFrom(&**it_1) && !(*it_2)->isSame(&**it_1))
```

Comment:
Please use `()` to make `&**it_1` more readable.

Reply:
Fixed.